import/index_java/src/org/solrmarc/index/VuFindIndexer.java

```diff
@@ -1444,35 +1444,59 @@ public static String getDeweySortableByType(
         if (list034 != null) {
             for (VariableField vf : list034) {
                 DataField df = (DataField) vf;
-                String d = df.getSubfield('d').getData();
-                String e = df.getSubfield('e').getData();
-                String f = df.getSubfield('f').getData();
-                String g = df.getSubfield('g').getData();
-                //System.out.println("raw Coords: "+d+" "+e+" "+f+" "+g);
-
-                // Check to see if there are only 2 coordinates
-                // If so, copy them into the corresponding coordinate fields
-                if ((d !=null && (e == null || e.trim().equals(""))) && (f != null && (g==null || g.trim().equals("")))) {
-                    e = d;
-                    g = f;
+                // Initialize coordinate variables
+                String d_coord = "";
+                String e_coord = "";
+                String f_coord = "";
+                String g_coord = "";
+
+                if (df.getSubfield('d') != null) {
```

Comment:
It might be slightly more efficient to replace this with something like:

<pre>
Subfield d_subfield = df.getSubfield('d');
if (d_subfield != null) {
    d_coord = d_subfield.getData();
}
</pre>

(to avoid doing the subfield lookup twice).

It might simplify the code even further to use a HashMap<Character, String> to map subfield to data value -- then you could do this retrieval in a loop instead of copying and pasting the logic multiple times. You could do something like:

<pre>
HashMap<Character, String> coords = new HashMap<Character, String>();
for (char code = 'd'; code <= 'g'; code++) {
    Subfield subfield = df.getSubfield(code);
    if (subfield != null) {
        coords.put(code, subfield.getData());
    }
}
</pre>

...instead of having four separate retrievals/validations/variables.

Then later in the code, instead of doing a null check, you can do `coords.containsKey('a')` to check if a value is set, and `coords.get('a')` to fetch the retrieved value.

I notice that there is a lot of logic duplicated between getAllCoordinates(), getDisplayCoordinates() and getPointCoordinates(). By using this HashMap approach, you might be able to deduplicate that logic, creating a method that takes in a data field and spits out a HashMap of validated/normalized coordinates, thus making all of those methods shorter and more readable.